packages/studio-plugin/src/createStudioStylingPlugin.ts

```diff
@@ -0,0 +1,58 @@
+import { PluginOption } from "vite";
+import upath from "upath";
+import { fileURLToPath } from "url";
+const currentDir = upath.dirname(fileURLToPath(import.meta.url));
+const packagesDir = upath.resolve(currentDir, "../..");
+
+/**
+ * Adds the ?inline query parameter to all CSS imports
+ * unless they are CSS from Studio, which is determined
+ * from the path of its importer. This prevents user styling
+ * from affecting Studio UI.
+ */
+export default function createStudioStylingPlugin(): PluginOption {
+  return {
+    name: "yext-studio-style-plugin",
+    enforce: "pre",
+    async resolveId(id, importer) {
+      if (!importer) {
+        return;
+      }
+      if (isImportedByStudio(importer)) {
+        return;
+      }
+      if (isStyleFile(id)) {
+        const resolvedId = (
+          await this.resolve(id, importer, { skipSelf: true })
```

Comment:
does this resolve the imported path to an absolute file path? and does the `skipSelf` skip running the plugin's `resolveId` logic on it while resolving?

Reply:
yep! returns an absolute path.  And yeah, that's exactly what`skipSelf` does.  It is supposed to default to `true` but I found that it doesn't.  Without that flag, it'll enter an infinite loop since it calls `resolveId` again. For more info if you're interested: https://rollupjs.org/plugin-development/#this-resolve